File: android/app/src/main/java/com/longforus/fPix/MyApplication.java
package com.longforus.fPix;

import android.app.Application;
import android.content.Context;
import com.idlefish.flutterboost.FlutterBoost;
import com.idlefish.flutterboost.Platform;
import com.idlefish.flutterboost.Utils;
import com.idlefish.flutterboost.interfaces.INativeRouter;
import io.flutter.embedding.android.FlutterView;
import java.util.Map;

public class MyApplication extends Application {


    @Override
    public void onCreate() {
        super.onCreate();

        INativeRouter router =new INativeRouter() {
            @Override
            public void openContainer(Context context, String url, Map<String, Object> urlParams, int requestCode, Map<String, Object> exts) {
               String  assembleUrl=Utils.assembleUrl(url,urlParams);
                PageRouter.openPageByUrl(context,assembleUrl, urlParams);
            }

        };

        FlutterBoost.BoostLifecycleListener boostLifecycleListener= new FlutterBoost.BoostLifecycleListener(){

            @Override
            public void beforeCreateEngine() {

            }

            @Override
            public void onEngineCreated() {

            }

            @Override
            public void onPluginsRegistered() {

            }

            @Override
            public void onEngineDestroy() {

            }

        };

        //
        // AndroidManifest.xml 中必须要添加 flutterEmbedding 版本设置
        //
        //   <meta-data android:name="flutterEmbedding"
        //               android:value="2">
        //    </meta-data>
        // GeneratedPluginRegistrant 会自动生成 新的插件方式　
        //
        // 插件注册方式请使用
        // FlutterBoost.instance().engineProvider().getPlugins().add(new FlutterPlugin());
        // GeneratedPluginRegistrant.registerWith()，是在engine 创建后马上执行，放射形式调用
        //

        Platform platform= new FlutterBoost
                .ConfigBuilder(this,router)
                .isDebug(true)
                .whenEngineStart(FlutterBoost.ConfigBuilder.ANY_ACTIVITY_CREATED)
                .renderMode(FlutterView.RenderMode.texture)
                .lifecycleListener(boostLifecycleListener)
                .build();
        FlutterBoost.instance().init(platform);



    }
}

File: android/app/src/main/kotlin/com/longforus/fPix/MyApplication.kt
package com.longforus.fPix

import android.app.Application
import android.util.Log
import com.idlefish.flutterboost.NewFlutterBoost
import com.idlefish.flutterboost.NewFlutterBoost.BoostLifecycleListener
import com.idlefish.flutterboost.NewFlutterBoost.ConfigBuilder
import com.idlefish.flutterboost.Utils
import com.idlefish.flutterboost.interfaces.INativeRouter
import io.flutter.embedding.android.FlutterView
import io.flutter.plugin.common.MethodChannel

class MyApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        val router = INativeRouter { context, url, urlParams, requestCode, exts ->
            val assembleUrl = Utils.assembleUrl(url, urlParams)
            PageRouter.openPageByUrl(context, assembleUrl, urlParams)
        }
        val lifecycleListener: BoostLifecycleListener = object : BoostLifecycleListener {
            override fun onEngineCreated() {}
            override fun onPluginsRegistered() {
                val mMethodChannel = MethodChannel(NewFlutterBoost.instance().engineProvider().dartExecutor, "methodChannel")
                Log.e("MyApplication", "MethodChannel create")
                //TextPlatformViewPlugin.register(NewFlutterBoost.instance().getPluginRegistry().registrarFor("TextPlatformViewPlugin"));
            }

            override fun onEngineDestroy() {}
        }
        val platform = ConfigBuilder(this, router)
            .isDebug(true)
            .whenEngineStart(ConfigBuilder.ANY_ACTIVITY_CREATED)
            .renderMode(FlutterView.RenderMode.texture)
            .lifecycleListener(lifecycleListener)
            .build()
        NewFlutterBoost.instance().init(platform)
    }
}